sklearn/feature_selection/tests/test_variance_threshold.py

```diff
@@ -1,33 +1,39 @@
 import numpy as np
 import pytest
-from scipy.sparse import bsr_matrix, csc_matrix, csr_matrix
 
 from sklearn.feature_selection import VarianceThreshold
 from sklearn.utils._testing import assert_array_equal
+from sklearn.utils.fixes import BSR_CONTAINERS, CSC_CONTAINERS, CSR_CONTAINERS
 
 data = [[0, 1, 2, 3, 4], [0, 2, 2, 3, 5], [1, 1, 2, 4, 0]]
 
 data2 = [[-0.13725701]] * 10
 
 
-def test_zero_variance():
+@pytest.mark.parametrize(
+    "constructor", [None, *BSR_CONTAINERS, *CSC_CONTAINERS, *CSR_CONTAINERS]
+)
+def test_zero_variance(constructor):
     # Test VarianceThreshold with default setting, zero variance.
+    X = data if constructor is None else constructor(data)
+    sel = VarianceThreshold().fit(X)
+    assert_array_equal([0, 1, 3, 4], sel.get_support(indices=True))
 
-    for X in [data, csr_matrix(data), csc_matrix(data), bsr_matrix(data)]:
-        sel = VarianceThreshold().fit(X)
-        assert_array_equal([0, 1, 3, 4], sel.get_support(indices=True))
 
+def test_zero_variance_value_error():
+    # Test VarianceThreshold with default setting, zero variance, error cases.
```

Comment:
I am not sure what this change is exactly since it seems unrelated.

Reply:
I'm splitting these into another test case because they are invariant of the container, and I think it is unnecessary to run the same cases multiple times.